azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/MavenFunctionContext.java

```diff
@@ -0,0 +1,149 @@
+package com.microsoft.azure.maven.function;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.project.MavenProject;
+
+import com.microsoft.azure.auth.configuration.AuthConfiguration;
+import com.microsoft.azure.common.docker.IDockerCredentialProvider;
+import com.microsoft.azure.common.function.IFunctionContext;
+import com.microsoft.azure.common.project.IProject;
+import com.microsoft.azure.common.project.JavaProject;
+import com.microsoft.azure.maven.function.configurations.RuntimeConfiguration;
+
+public class MavenFunctionContext implements IFunctionContext {
+	private AbstractFunctionMojo mojo;
+	private IProject project;
+	private Map<Class<? extends Object>, Object> providerMap = new HashMap<>();
+
+	public MavenFunctionContext(AbstractFunctionMojo mojo) {
+		this.mojo = mojo;
+		this.project = convertProject(mojo.getProject());
+	}
+
+	@Override
+	public String getDeploymentStagingDirectoryPath() {
+		return mojo.getDeploymentStagingDirectoryPath();
+	}
+
+	@Override
+	public String getSubscription() {
+		return mojo.getSubscriptionId();
+	}
+
+	@Override
+	public String getAppName() {
+		return mojo.getAppName();
+	}
+
+	@Override
+	public String getResourceGroup() {
+		return mojo.getResourceGroup();
+	}
+
+	@Override
+	public RuntimeConfiguration getRuntime() {
+		return mojo.getRuntime();
+	}
+
+	@Override
+	public String getRegion() {
+		return mojo.getRegion();
+	}
+
+	@Override
+	public String getPricingTier() {
+		return mojo.getPricingTier();
+	}
+
+	@Override
+	public String getAppServicePlanResourceGroup() {
+		return mojo.getAppServicePlanResourceGroup();
+	}
+
+	@Override
+	public String getAppServicePlanName() {
+		return mojo.getAppServicePlanName();
+	}
+
+	@Override
+	public Map getAppSettings() {
+		return mojo.getAppSettings();
+	}
+
+	@Override
+	public AuthConfiguration getAuth() {
+		return null;
+	}
+
+	@Override
+	public String getDeploymentType() {
+		return mojo.getDeploymentType();
+	}
+
+	public IDockerCredentialProvider getDockerCredentialProvider() {
+		return getProvider(IDockerCredentialProvider.class);
+	}
+
+	@Override
+	public void registerProvider(Class<? extends Object> clazz, Object provider) {
```

Comment:
clazz->class?

Reply:
class is keyword

Reply:
so may be outputClass or some more meaningful name?